frontend/cypress/integration/common/graph_context_menu.ts

```diff
@@ -44,3 +84,45 @@ Then('user should see {string} cluster parameter in links in the context menu',
     clusterParameterExists(present);
   });
 });
+
+Then(
+  'user should see the {string} cluster parameter in the {string} link in the context menu',
+  (cluster: string, linkText: string) => {
+    cy.get(`[data-test="graph-node-context-menu"]`).within(() => {
+      cy.get('a').contains(linkText).should('have.attr', 'href').and('include', `clusterName=${cluster}`);
+    });
+  }
+);
+
+// @istio-config-cleanup
+After({ tags: '@istio-config-cleanup' }, () => {
```

Comment:
I love the idea, but I have a few suggestions:

It is usually better to perform cleanups optionally before the execution of a specific test and not after. The remaining objects can be still handy for debugging purposes.

The name of the tag seems too general, as it does not cover all of the Istio configs. 

I would also move this hook to the hooks.ts file.

Reply:
> It is usually better to perform cleanups optionally before the execution of a specific test and not after. The remaining objects can be still handy for debugging purposes.

This is true but keeping these objects around affects actual traffic on the graph which affects the rest of the tests. With multi-primary, you actually need to setup the request routing for each cluster that your service is located in otherwise traffic won't be sent to the service located in the cluster without the request routing objects created there. At first I tried duplicating the objects to both clusters but this was more awkward than just deleting the objects right after the test completed. I can give that another try.

Reply:
Alright now the configuration gets duplicated to both clusters as part of the scenario rather than removing the istio config at the end.